hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/HazelcastTypeCoercion.java

```diff
@@ -94,6 +95,22 @@ private boolean coerceNode(
         return true;
     }
 
+    private static SqlNode cast(SqlNode node, SqlDataTypeSpec targetTypeSpec) {
+        if (node.getKind() == SqlKind.ARGUMENT_ASSIGNMENT) {
+            // transform name => 'value' into name => cast('value')
+            SqlBasicCall call = (SqlBasicCall) node;
+
+            SqlNode value = call.getOperandList().get(0);
+            SqlNode name = call.getOperandList().get(1);
+
+            SqlNode cast = HazelcastSqlOperatorTable.CAST.createCall(SqlParserPos.ZERO, value, targetTypeSpec);
```

Comment:
We can do a recursive call:
```suggestion
            SqlNode cast = cast(value, targetTypeSpec);
```